### PR TITLE
comments for K8 describe package

### LINKED ISSUE #287: Writing Unit tests For k8 in Utils package
I  was thinking if its possible to write unit tests for each of the  [Kubernetes](https://github.com/meshery/meshkit/tree/master/utils/kubernetes) package code files. 
If it is i would like to engage in this

### PR DESCRIPTION
**Pull request by Philip-21**
Signed-off-by: Philip-21 <philipuzomaobiora@gmail.com>
**Description**
Created comments for k8 describe package

This PR fixes #287 

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by Aisuko — CHANGES_REQUESTED** [review]
Can we only include the code comments for `describe.go`? Before we confirm the scope of the mocking test, we could merge that test case

**Comment by Philip-21** [comment]
@Aisuko from what you are saying, do you want me to only comment on the role for the describe.go  so it could tally with describe_test.go and should  i also take out some of the comments i wrote in the describe.go ?

**Comment by Aisuko** [comment]
> @Aisuko from what you are saying, do you want me to only comment on the role for the describe.go so it could tally with describe_test.go and should I also take out some of the comments i wrote for the codes in the describe.go ?

Please keep the code comments in the `descrble.go`. The mock testing part already include in your previous PR right? This PR is only for code comments. So, we can merge it directly.

**Comment by Aisuko** [comment]
Please do not forget to update to the latest version code.

**Comment by Philip-21** [comment]
> > @Aisuko from what you are saying, do you want me to only comment on the role for the describe.go so it could tally with describe_test.go and should I also take out some of the comments i wrote for the codes in the describe.go ?
> 
> Please keep the code comments in the `descrble.go`. The mock testing part already include in your previous PR right? This PR is only for code comments. So, we can merge it directly.

Yes this pr only includes code comments for the describe.go . The other Pr is for the Mock testing

**Comment by Philip-21** [comment]
> Please do not forget to update to the latest version code.

Please how do you mean by this

**Comment by Aisuko** [comment]
Screenshot 2023-04-12 at 11 48 50 am

**Comment by Philip-21** [comment]
> Screenshot 2023-04-12 at 11 48 50 am

ok thanks,  i  have just updated branch to latest version.

**Comment by Philip-21** [comment]
> We need to only include the `describe.go` file in this PR. This PR only aims to add code comment to `describle.go`

ok i have removed the describe_test.go file . The describe.go file is what is in this pr

File: utils/kubernetes/describe/describe.go [hunk 1]
```diff
@@ -17,13 +19,18 @@ import (
...
-	ChunkSize  int64
-	Type       DescribeType
+	Name       string       // Name of the kubernetes obj
+	Namespace  string       // Namespace of the kubernetes obj
+	ShowEvents bool         // A boolean flag indicating whether to show events associated with the Kubernetes object or not.
+	ChunkSize  int64        //Size of the chunk in which the Kubernetes object's output is written.
+	Type       DescribeType //an integer value that represents the Kubernetes source that needs to be described
 }
 
+/*
```
Comment (Aisuko) [inline utils/kubernetes/describe/describe.go:29]: Replace to `// DescribeType is an integer value that represents the Kubernetes resource that needs to be described.`

File: utils/kubernetes/describe/describe.go [hunk 2]
```diff
@@ -57,6 +64,12 @@ const (
 	EndpointSlice
 )
 
+/*
```
Comment (Aisuko) [inline utils/kubernetes/describe/describe.go:67]: Replace to `// The ResourceMap variable contains the GroupKind information of all the Kubernetes resources that can be described.`

File: utils/kubernetes/describe/describe.go [hunk 3]
```diff
@@ -88,22 +101,30 @@ var ResourceMap = map[DescribeType]schema.GroupKind{
 	EndpointSlice:             {Group: discoveryv1.GroupName, Kind: "EndpointSlice"},
 }
 
+/*
```
Comment (Aisuko) [inline utils/kubernetes/describe/describe.go:104]: Replace to `// The Describe() function takes a meshkitkube.Client object and a DescriberOptions object as input.
// And it returns the description of the specified Kubernetes resource as a string.`

File: utils/kubernetes/describe/describe.go [hunk 4]
```diff
@@ -88,22 +101,30 @@ var ResourceMap = map[DescribeType]schema.GroupKind{
...
 		ShowEvents: options.ShowEvents,
 		ChunkSize:  options.ChunkSize,
 	}
+	//calls a corresponding "describer" function to retrieve the description of the specified Kubernetes resource
 	output, err := describer.Describe(options.Namespace, options.Name, describerSetting)
 	if err != nil {
 		return "", err
 	}
-
+	/*
```
Comment (Aisuko) [inline utils/kubernetes/describe/describe.go:125]: Please remove this code comment.

**Review by Aisuko — CHANGES_REQUESTED** [review]
Thank you, please take a look and help me replace these code comments. We use code comment `//` in the file.

File: utils/kubernetes/describe/describe.go (hunk 4 above)
Comment (Philip-21) [reply utils/kubernetes/describe/describe.go]: Ok, i have made the changes

File: utils/kubernetes/describe/describe.go [hunk 5]
```diff
@@ -17,13 +19,14 @@ import (
...
-	ChunkSize  int64
-	Type       DescribeType
+	Name       string       // Name of the kubernetes obj
+	Namespace  string       // Namespace of the kubernetes obj
+	ShowEvents bool         // A boolean flag indicating whether to show events associated with the Kubernetes object or not.
+	ChunkSize  int64        //Size of the chunk in which the Kubernetes object's output is written.
+	Type       DescribeType //an integer value that represents the Kubernetes source that needs to be described
 }
 
+ // DescribeType is an integer value that represents the Kubernetes resource that needs to be described.
```
Comment (Aisuko) [inline utils/kubernetes/describe/describe.go:29]: Here is a blank at the beginning, CI cannot run succeed. So, please remove it.

File: utils/kubernetes/describe/describe.go [hunk 6]
```diff
@@ -57,6 +60,7 @@ const (
 	EndpointSlice
 )
 
+ // The ResourceMap variable contains the GroupKind information of all the Kubernetes resources that can be described.
```
Comment (Aisuko) [inline utils/kubernetes/describe/describe.go:63]: Here.

File: utils/kubernetes/describe/describe.go [hunk 7]
```diff
@@ -88,22 +92,23 @@ var ResourceMap = map[DescribeType]schema.GroupKind{
 	EndpointSlice:             {Group: discoveryv1.GroupName, Kind: "EndpointSlice"},
 }
 
+// The Describe() function takes a meshkitkube.Client object and a DescriberOptions object as input.
+ // And it returns the description of the specified Kubernetes resource as a string.
```
Comment (Aisuko) [inline utils/kubernetes/describe/describe.go:96]: Here.

**Review by Aisuko — CHANGES_REQUESTED** [review]
After remove these blanks and pass CI. We can merge it. You can get more detail from https://stackoverflow.com/questions/69049182/file-is-not-gofmt-ed-with-s-why-is-this-happening-and-how-to-resolve-it

File: utils/kubernetes/describe/describe.go (hunk 7 above)
Comment (Philip-21) [reply utils/kubernetes/describe/describe.go:96]: You  want me to remove the comments or the spaces in between the comments ?

Comment (Aisuko) [reply utils/kubernetes/describe/describe.go:96]: The space at the second comment

Comment (Philip-21) [reply utils/kubernetes/describe/describe.go]: i have removed the spaces

**Comment by Aisuko** [comment]
CI may have some issues, I check the PR manually by using golangci-lint, there is no error. `ErrGetDescriberFunc` was defined under the same package.

Screenshot 2023-04-13 at 6 25 40 pm

**Review by Aisuko — APPROVED** [review]
LGTM

**Comment by Philip-21** [comment]
> CI may have some issues, I check the PR manually by using golangci-lint, there is no error. `ErrGetDescriberFunc` was defined under the same package.
> 
> Screenshot 2023-04-13 at 6 25 40 pm

does it mean that it shouldn't be there .

**Comment by Aisuko** [comment]
> > CI may have some issues, I check the PR manually by using golangci-lint, there is no error. `ErrGetDescriberFunc` was defined under the same package.
> > Screenshot 2023-04-13 at 6 25 40 pm
> 
> does it mean that it shouldn't be there .

No worries, this is no related to this PR.

**Pull request merged**